### ISSUE TITLE
Weights for FSC-147

### ISSUE THREAD

**Issue by Muhammad-Ibraheem-Siddiqui** [Ryh1218/FSOL#1]
Hi,

Thanks for the great work. Can you share the training checkpoints for FSC-147 dataset? I have tried to train on single GPU considering FSC-147 dataset for 200 epochs but the results are not as reported in the paper. Thanks in advance.

**Comment by Ryh1218** [comment]
Thanks for your attention. The pretrained weight of FSOL has uploaded along with how to load them:)

**Comment by Muhammad-Ibraheem-Siddiqui** [comment]
Thanks for the prompt response. There is no weight file in the uploaded content. Screenshots are attached herewith. Kindly let me know if I am missing something. Thanks

![image](https://github.com/user-attachments/assets/bad39c82-7380-4b3a-b065-a52e2fb90d3e)
![image](https://github.com/user-attachments/assets/e8357675-f590-4b09-9de8-b22615c030d3)

**Comment by Muhammad-Ibraheem-Siddiqui** [comment]
Thanks for the prompt response. While testing on FSC-147 using the provided weights, I am getting MAE = 18.6 and MSE = 104.6 in the one-shot case (screenshot attached). This does not coincide with the results stated in your paper i.e, MAE = 20.33 and MSE =68.28. Can you explain this change or is there something else I am missing? Thanks. 

![image](https://github.com/user-attachments/assets/75c48ffc-0588-4677-8070-38804d235d83)

**Comment by Muhammad-Ibraheem-Siddiqui** [comment]
I just realized that it is the results on the evaluation dataset that you have reported in the paper for the FSC-147, one-shot case. I managed to replicate the results. Can you comment about the performance on the test set as mentioned in the comment above? Thanks.

**Comment by Ryh1218** [comment]
Sorry for the late response. In our paper, we report the results on the evaluation set. In the ablation study, since our FSOL model is a localization model, we skip the counting performance and focus on the localization metrics. Your results are similar to our recently attached reproduction results.

Regarding the difference in counting performance between the validation set and the test set, please note that in the FSC-147 dataset, the categories in the validation set and the test set are disjoint, while the FSOL model does not require test-time adaptation. So, it is natural to expect differences.
![image](https://github.com/user-attachments/assets/ae50d21f-4dab-4b6b-8e64-07470bc6effa)

**Comment by huytuong010101** [comment]
@Ryh1218 Hi, thank you for a perfect repository.
Can you share the code for prediction? I just find the code for training purpose in your repo
Thank you

**Comment by Ryh1218** [comment]
@huytuong010101 Thank you for your comment!

To test the model, after loading the trained weights, you can run 'sh eval.sh' from the 'experiments/{experimenting_dataset}/' directory. Please note that, according to the dataset configurations, only the FSC-147 dataset includes a predefined train/validation/test split. Therefore, you can run 'sh test.sh' only for the FSC-147 dataset.

Thanks again!
